control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1615,56 +1639,68 @@ func setStatusConditionWithTransitionUpdate(conditions *[]metav1.Condition, newC
 	existingCondition.ObservedGeneration = newCondition.ObservedGeneration
 }
 
-func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyperv1.HostedControlPlane) (sets.String, error) {
+func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyperv1.HostedControlPlane) (map[string]bool, error) {
```

Comment:
can you explain why is the signature of this function changing and comment the function with what it actually does?

Reply:
Previous this PR, sets were enough because the controller does not care about to ignore a resource, now introducing this map, the controller can understand when a objects is still trying to be deleted and the timeout wasn't reached yet vs when the controller already give up on a concrete resource deletion and could continue with deprovision.